### ISSUE TITLE
TSQL: Computed Columns in Table Variables

### ISSUE THREAD

**Issue by ddbtrmatic** [sqlfluff/sqlfluff#7166]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Computed columns in table variables result in parsing errors.

### Expected Behaviour

TSQL supports computed columns defined in table variables. These should be parsed and formatted just as a normal table definition.

Related to: https://github.com/sqlfluff/sqlfluff/pull/5792

### Observed Behaviour

Parsing errors. See reproduction below.

### How to reproduce

```sql

    DECLARE
        @myTable TABLE
        (
            ID INT,
            MyCol1 BIT,
            MyCol2 BIT,
            MyComputedCol AS (
                CASE
                    WHEN MyCol1 & MyCol2 = 0
                    THEN 1
                    WHEN MyCol2 = 0 THEN 2
                    ELSE 3
                END
            )
        );
```

```sh
sqlfluff fix lint.sql
==== finding fixable violations ====
== [lint.sql] FAIL
L:   1 | P:   1 | LT01 | Expected only single space before word. Found '    '.
                       | [layout.spacing]
L:   1 | P:   1 | LT02 | First line should not be indented. [layout.indent]
L:   1 | P:   1 | LT13 | Files must not begin with newlines or whitespace.
                       | [layout.start_of_file]
==== no fixable linting violations found ====
All Finished!
  [3 unfixable linting violations found]
  [1 templating/parsing errors found]

Process finished with exit code 1
```

Removing the computed column:

```sql

    DECLARE
        @myTable TABLE
        (
            ID INT,
            MyCol1 BIT,
            MyCol2 BIT
        );
```

```sh
sqlfluff fix lint.sql
==== finding fixable violations ====
== [lint.sql] FAIL
L:   1 | P:   1 | LT01 | Expected only single space before 'DECLARE' keyword.
                       | Found '    '. [layout.spacing]
L:   1 | P:   1 | LT02 | First line should not be indented. [layout.indent]
L:   2 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:   3 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:   4 | P:   1 | LT02 | Expected indent of 8 spaces. [layout.indent]
L:   5 | P:   1 | LT02 | Expected indent of 8 spaces. [layout.indent]
L:   6 | P:   1 | LT02 | Expected indent of 8 spaces. [layout.indent]
L:   7 | P:   1 | LT02 | Expected indent of 4 spaces. [layout.indent]
L:   7 | P:  10 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
== [lint.sql] FIXED
9 fixable linting violations found

Process finished with exit code 0
```

### Dialect

TSQL

### Version

sqlfluff, version 3.4.2
Python 3.13.7

### Configuration

```
[sqlfluff]
dialect = tsql
large_file_skip_byte_limit = 200000
encoding = utf-8
max_line_length = 120
exclude_rules = AL03,capitalisation.identifiers
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper
[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = pascal
[sqlfluff:rules:references.special_chars]
additional_allowed_characters = '.@'
[sqlfluff:rules:convention.not_equal]
preferred_not_equal_style = ansi
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
